node/src/block_producer/block_producer_reducer.rs

```diff
@@ -353,6 +337,11 @@ impl BlockProducerEnabled {
                         }
                     )),
                 });
+
+                #[cfg(feature = "p2p-libp2p")]
+                dispatcher.push(p2p::P2pNetworkPubsubAction::BroadcastBlock {
```

Comment:
@vlad9486 @binier  is this needed? shouldn't the best tip update logic itself handle it? otherwise how do we broadcast blocks that we didn't produce?

Reply:
And if it is really needed, I would move it to `BlockInjected` bellow, we want to make sure we succeeded before broadcasting it right?